.github/workflows/pdf-build.yaml

```diff
@@ -87,8 +67,46 @@ jobs:
             echo "ARTIFACT_NAME=economy_printable_${{ inputs.language }}" >> $GITHUB_ENV
           fi
 
+      - name: Cache built PDF
+        uses: actions/cache@v4
+        if: github.event_name == 'pull_request'
+        id: cache-pdf
+        with:
+          path: |
+            ./${{ env.OUTPUT_FILENAME }}
+          key: >-
+            pdf-${{ inputs.build_type }}-${{ inputs.language }}-${{
+            inputs.language == 'en'
+            && hashFiles('main_en.tex', 'metadata.tex', 'sections/*.tex', 'assets/**/*', 'latexmkrc', 'index_style.ist')
+            || hashFiles('main_${{ inputs.language }}.tex', 'metadata.tex', format('sections/translated/{0}/*.tex', inputs.language), 'assets/**/*', format('translations/assets/{0}/**/*', inputs.language), 'latexmkrc', 'index_style.ist')
+            }}
+          restore-keys: |
+            pdf-${{ inputs.build_type }}-${{ inputs.language }}-
+
+      - name: Cache LaTeX packages and build
+        uses: actions/cache@v4
+        if: steps.cache-pdf.outputs.cache-hit != 'true'
```

Comment:
shouldn't the step below be using the id of this step instead?

Reply:
This step caches only latexmk artifacts, so it does not replace the build, only speeds it up a little. It shouldn't affect whether the build runs or not.